packages/flutter_tools/lib/src/build_system/targets/darwin.dart

```diff
@@ -0,0 +1,147 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:meta/meta.dart';
+
+import '../../base/common.dart';
+import '../../base/file_system.dart';
+import '../../build_info.dart';
+import '../../flutter_plugins.dart';
+import '../../globals.dart' as globals;
+import '../build_system.dart';
+import '../exceptions.dart';
+
+/// A target that checks that dev dependencies are enabled on debug builds and
+/// disabled on release/profile builds.
+///
+/// The Flutter tool enables/disables dev dependencies using the build mode.
+/// These can get out of sync if the user switches the build mode in Xcode.
+///
+/// Implementers should override [CheckDevDependencies.inputs] to add a [Source]
+/// that changes when dev dependencies status changes.
+abstract class CheckDevDependencies extends Target {
+  const CheckDevDependencies();
+
+  @override
+  List<Target> get dependencies => <Target>[];
+
+  @override
+  List<Source> get inputs => <Source>[
+    const Source.pattern(
+      '{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/darwin.dart',
+    ),
+    const Source.pattern('{PROJECT_DIR}/.flutter-plugins-dependencies'),
```

Comment:
Does it break if there is no `.flutter-plugins-dependencies`? I believe `.flutter-plugins-dependencies` is only generated if there are plugins.

Nit: Also, I'd avoid using a hardcoded path, since it could change. 

```suggestion
     final FlutterProject flutterProject = FlutterProject.current();
    final String pluginsDependenciesFilePath = flutterProject.flutterPluginsDependenciesFile.path.replaceFirst(
      flutterProject.directory.path,
      '{PROJECT_DIR}/',
    );
   return <Source>[
    const Source.pattern(
      '{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/darwin.dart',
    ),
    const Source.pattern(pluginsDependenciesFilePath),
```

Reply:
> Does it break if there is no `.flutter-plugins-dependencies`? I believe `.flutter-plugins-dependencies` is only generated if there are plugins.

It doesn't break but it does cause the target to run on every build since the input is missing.

> Nit: Also, I'd avoid using a hardcoded path, since it could change.

This is a good idea, but since this is a common pattern I added a `Source` factory to make this cleaner. For example:

```dart
Source.fromProject((FlutterProject project) => project.fooFile);
```